DataHolderLib/Project.cpp

```diff
@@ -0,0 +1,116 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include "Project.h"
+
+#include <algorithm>
+
+// ThirdParty/logog
+#include <logog/include/logog.hpp>
+
+#include "BaseLib/FileTools.h"
+#include "BaseLib/uniqueInsert.h"
+
+#include "MeshLib/Mesh.h"
+
+namespace DataHolderLib
+{
+
+Project::~Project()
+{
+	for (MeshLib::Mesh* m : _mesh_vec)
+		delete m;
+}
+
+void Project::addMesh(MeshLib::Mesh* mesh)
+{
+	std::string name = mesh->getName();
+	getUniqueName(name);
+	mesh->setName(name);
+	_mesh_vec.push_back(mesh);
+}
+
+std::vector<MeshLib::Mesh*>::const_iterator Project::findMeshByName(
+		std::string const& name) const
+{
+	return const_cast<Project&>(*this).findMeshByName(name);
+}
+
+std::vector<MeshLib::Mesh*>::iterator Project::findMeshByName(
+		std::string const& name)
+{
+	return std::find_if(_mesh_vec.begin(), _mesh_vec.end(),
+			[&name](MeshLib::Mesh* mesh)
+			{
+				return mesh && (name == mesh->getName());
+			});
+}
+
+const MeshLib::Mesh* Project::getMesh(const std::string &name) const
+{
+	std::vector<MeshLib::Mesh*>::const_iterator it = findMeshByName(name);
+	return (it == _mesh_vec.end() ? nullptr : *it);
+}
+
+bool Project::removeMesh(const std::string &name)
+{
+	bool mesh_found = false;
+	std::vector<MeshLib::Mesh*>::iterator it = findMeshByName(name);
+	while (it != _mesh_vec.end())
+	{
+		delete *it;
+		*it = nullptr;
+		it = findMeshByName(name);
+		mesh_found = true;
+	}
+
+	_mesh_vec.erase(std::remove(_mesh_vec.begin(), _mesh_vec.end(), nullptr),
```

Comment:
I wonder why the entry isn't erased above in line 69? Maybe we discuss this tomorrow?!